tests/networks/networks_test.py

```diff
@@ -0,0 +1,77 @@
+# python3
+# Copyright 2021 InstaDeep Ltd. All rights reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""Example running MADDPG on debug MPE environments."""
+from absl import flags
+
+from mava import specs as mava_specs
+from mava.systems.tf import mad4pg, maddpg, madqn, mappo, qmix, vdn
+from mava.utils.environments import debugging_utils
+
+FLAGS = flags.FLAGS
+flags.DEFINE_string(
+    "env_name",
+    "simple_spread",
+    "Debugging environment name (str).",
+)
+flags.DEFINE_string(
+    "action_space",
+    "discrete",
+    "Environment action space type (str).",
+)
+
+
+class TestNetworks:
+    # Test that we can load a env module and that it contains agents,
+    #   agents and possible_agents.
+    def test_network_keys(self) -> None:
+        # Environment.
+        env = debugging_utils.make_environment(
+            env_name=FLAGS.env_name, action_space=FLAGS.action_space, evaluation=False
+        )
+        environment_spec = mava_specs.MAEnvironmentSpec(env)
+
+        system_fns = [maddpg, mad4pg, madqn, mappo, qmix, vdn]
+
+        for system in system_fns:
+            # Test shared weights setup.
+            agent_net_keys = {
+                "agent_0": "agent",
+                "agent_1": "agent",
+                "agent_2": "agent",
+            }
+            networks = system.make_default_networks(  # type: ignore
+                environment_spec=environment_spec,  # type: ignore
+                agent_net_keys=agent_net_keys,
+            )
+
+            for key in networks.keys():
+                assert ["agent"] == list(networks[key].keys())
```

Comment:
I think we should perhaps make the two asserts (one here and below) two separate tests? Also, this test is nice for checking that the key strings much up, but should we also add a test to check the actual difference in the weights in the networks? Or do you think it is sufficient to just check the keys?

Reply:
I think two tests might also be better. Incase it fails, easier to know which is one failed and no dependency between the two. 

@DriesSmit  Are we still adding the tests that check the networks at a system level? I think that answers @arnupretorius 's questions.

Reply:
@arnupretorius I changed it to be two tests. I don't think we can test if the variables are shared here? Variables only get initialised inside the builder. Let me know if that answers your question?

Reply:
> I think two tests might also be better. Incase it fails, easier to know which is one failed and no dependency between the two.
> 
> @DriesSmit Are we still adding the tests that check the networks at a system level? I think that answers @arnupretorius 's questions.

@KaleabTessera We can still add system-level tests yes. Let me know if you still think it is needed :+1:

Reply:
@DriesSmit , I think we should be able to test it in a similar way to how we do the smoke tests for the systems, i.e. write the test once the variables are already initialised. I think this is what @KaleabTessera is referring to.

Reply:
For example, here: https://github.com/instadeepai/Mava/blob/develop/tests/systems/maddpg_system_test.py